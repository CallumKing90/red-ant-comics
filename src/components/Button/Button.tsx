import React from "react";

interface ButtonProps {
  className?: string;
  children?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

export function Button({
  className,
  children,
  onClick,
}: ButtonProps): JSX.Element {
  return (
    <button className={className} onClick={onClick} role="button">
      {children}
    </button>
  );
}
