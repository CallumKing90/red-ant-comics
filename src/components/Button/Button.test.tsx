import React from "react";
import { Button } from "./Button";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Button", () => {
  it("should render the button", () => {
    const { getByText } = render(<Button>Button text</Button>);

    const button = getByText("Button text");
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });

  it("should render with a class name", () => {
    const { getByText } = render(
      <Button className="className">Button text</Button>
    );

    const button = getByText("Button text");
    expect(button).toHaveClass("className");
    expect(button).toMatchSnapshot();
  });

  it("should render with an onClick function", () => {
    const testFn = jest.fn();
    const { getByText } = render(<Button onClick={testFn}>Button text</Button>);

    const button = getByText("Button text");
    expect(testFn).toBeCalledTimes(0);
    fireEvent.click(button);
    expect(testFn).toBeCalledTimes(1);
    expect(button).toMatchSnapshot();
  });
});
