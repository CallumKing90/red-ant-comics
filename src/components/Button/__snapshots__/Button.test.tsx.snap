// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<button
  role="button"
>
  Button text
</button>
`;

exports[`Button should render with a class name 1`] = `
<button
  class="className"
  role="button"
>
  Button text
</button>
`;

exports[`Button should render with an onClick function 1`] = `
<button
  role="button"
>
  Button text
</button>
`;
